src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -209,3 +230,35 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 register_cli_argument('vmss create', 'disable_overprovision', help='Overprovision option (see https://azure.microsoft.com/en-us/documentation/articles/virtual-machine-scale-sets-overview/ for details).', action='store_true')
 register_cli_argument('vmss create', 'upgrade_policy_mode', help=None, **enum_choice_list(UpgradeMode))
 register_cli_argument('vmss create', 'vm_sku', help='Size of VMs in the scale set.  See https://azure.microsoft.com/en-us/pricing/details/virtual-machines/ for size info.')
+
+existing_disk_name = CliArgumentType(overrides=name_arg_type, help='The name of the managed disk', completer=get_resource_name_completion_list('Microsoft.Compute/disks'), id_part='name')
+register_cli_argument('disk', 'disk_name', existing_disk_name, completer=get_resource_name_completion_list('Microsoft.Compute/disks'))
+register_cli_argument('disk', 'name', arg_type=name_arg_type)
+register_cli_argument('disk', 'storage_account_type', arg_type=disk_storage_account_type)
+
+existing_snapshot_name = CliArgumentType(overrides=name_arg_type, help='The name of the snapshot', completer=get_resource_name_completion_list('Microsoft.Compute/snapshots'), id_part='name')
+register_cli_argument('snapshot', 'snapshot_name', existing_snapshot_name, id_part='name', completer=get_resource_name_completion_list('Microsoft.Compute/snapshots'))
+register_cli_argument('snapshot', 'name', arg_type=name_arg_type)
+register_cli_argument('snapshot', 'storage_account_type', arg_type=disk_storage_account_type)
+
+existing_image_name = CliArgumentType(overrides=name_arg_type, help='The name of the custom image', completer=get_resource_name_completion_list('Microsoft.Compute/images'), id_part='name')
+register_cli_argument('image', 'os_type', **enum_choice_list(['Windows', 'Linux']))
+register_cli_argument('image', 'image_name', arg_type=name_arg_type, id_part='name', completer=get_resource_name_completion_list('Microsoft.Compute/images'))
+register_cli_argument('image create', 'name', arg_type=name_arg_type, validator=process_image_create_namespace, help='new image name')
+register_cli_argument('image create', 'os_blob_uri', help='blob uri of os virtual hard disk')
+register_cli_argument('image create', 'data_blob_uris', arg_type=multi_ids_type, help='space separated blob uris of data virtual hard disk')
+register_cli_argument('image create', 'os_snapshot', help='snapshot id or name to copy for the os disk')
+register_cli_argument('image create', 'os_disk', help='managed os disk id or name to copy for the os disk')
+register_cli_argument('image create', 'data_disks', arg_type=multi_ids_type, help='space separated managed disk id or name to copy for datat disks')
+register_cli_argument('image create', 'data_snapshots', arg_type=multi_ids_type, help='space separated snapshot id or name to copy for datat disks')
+register_cli_argument('image create', 'source_virtual_machine', help='id or name of a generalized virtual machine to capture image from',
```

Comment:
Correct, this is the offer from the managed-disk world for capturing generalized vm. 
To capture specialized vm, for now,  you can use `az disk create` and supply the name/id to the `az vm create --managed-os-disk`.  In around late February, we will provide finer control through VM back-up and restoration-points.